src/renderers/WebGLRenderer.js

```diff
@@ -1438,11 +1442,20 @@ function WebGLRenderer( parameters ) {
 
 		}
 
-		const progUniforms = materialProperties.program.getUniforms(),
-			uniformsList =
-				WebGLUniforms.seqWithValue( progUniforms.seq, uniforms );
+		materialProperties.uniformsList = null;
```

Comment:
I guess this line can be removed and in `getUniformList()` you check for `undefined`?

Reply:
Turns out this is the thing that broke the lightning strike example. I think the issue is that if some aspect of the material's uniform count changes (like the number of lights in a scene) the `uniformsList` needs to be explicitly cleared and rebuilt on next use.